mod/nodes/src/create_stream_action.go

```diff
@@ -0,0 +1,124 @@
+package nodes
+
+import (
+	"strings"
+
+	"github.com/cryptopunkscc/astrald/astral"
+	"github.com/cryptopunkscc/astrald/mod/exonet"
+	"github.com/cryptopunkscc/astrald/mod/nodes"
+	"github.com/cryptopunkscc/astrald/mod/scheduler"
+)
+
+var _ scheduler.Action = &CreateStreamAction{}
+
+type CreateStreamAction struct {
+	mod      *Module
+	Target   string
+	Net      string // optional
+	Endpoint string // optional in format "net:address"
```

Comment:
the action should take a slice of exonet.Endpoint to be used and how those endpoints are selected should remain as the responsibility of the caller (OpNewStream in this case). If we move selection logic to CreateStreamAction we make it impossible to feed to it a list of endpoints selected by different criteria.